src/pages/user/components/ReportModal.tsx

```diff
@@ -0,0 +1,139 @@
+import {
+  Button,
+  Checkbox,
+  Flex,
+  FormControl,
+  FormHelperText,
+  FormLabel,
+  Input,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Text,
+  VStack,
+  useDisclosure,
+  useMediaQuery
+} from '@chakra-ui/react'
+import { ChangeEvent, useRef, useState } from 'react'
+import { useAuthContext } from '../../../api/contexts/auth/useAuthContext'
+import { ReportDateRange, useAdminReportMutation, useUserReportMutation } from '../../../api/hooks/userMutationHooks'
+import { DownloadFileFromServerButton } from '../../../components/commons/DownloadFileFromServerButton'
+import { formatDate, getEndOfSemester, getStartOfSemester } from '../../../util/dateHelper'
+
+export const ReportModal = () => {
+  const { loggedInUser } = useAuthContext()
+  const [largeScreen] = useMediaQuery('(min-width: 48em)')
+  const { isOpen, onOpen, onClose } = useDisclosure()
+  const [startDate, setStartDate] = useState(getStartOfSemester())
+  const [endDate, setEndDate] = useState(getEndOfSemester())
+  const [adminReport, setAdminReport] = useState(false)
+
+  const getUserReportMutation = useUserReportMutation()
+  const userReportRef = useRef<HTMLButtonElement>(null)
+  const getAdminReportMutation = useAdminReportMutation()
+  const adminReportRef = useRef<HTMLButtonElement>(null)
+
+  const startDateError = startDate >= endDate
+  const endDateError = endDate > new Date()
+
+  const onEndDateChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const date = new Date(e.target.value)
+    date.setHours(23, 59, 59, 999)
+    setEndDate(date)
+  }
+
+  return (
+    <>
+      <Button colorScheme="green" onClick={onOpen}>
+        Riport generálása
+      </Button>
+      <Modal isCentered={largeScreen} motionPreset="slideInBottom" isOpen={isOpen} onClose={onClose}>
+        <ModalOverlay />
+        <ModalContent>
+          <ModalHeader>Riport generálása</ModalHeader>
+          <ModalCloseButton />
+          <ModalBody pb={6}>
+            <VStack alignItems="flex-start">
+              <Text>HK-s ösztöndíjakhoz itt tudsz riport generálni az általad tartott konzultációkról.</Text>
+              {loggedInUser?.isAdmin && (
```

Comment:
const isAdmin = loggedInUser?.isAdmin
would be nicer, it is used in multiple places